pilot/pkg/model/jwks_resolver.go

```diff
@@ -50,14 +50,21 @@ const (
 
 	// JwtPubKeyExpireDuration is the expire duration for JWT public key in the cache.
 	// After this duration expire, refresher job will fetch key for the cached item again.
-	JwtPubKeyExpireDuration = time.Hour
+	JwtPubKeyExpireDuration = 24 * 7 * time.Hour
```

Comment:
why change expire duration ?  though pubkey is relatively stable, making it shorter allow to detect change faster

Reply:
This is for the situation when the network is inaccessible for more than 1 hour, basically this means we will try to use the cached public key in this duration, I updated the refresh job to not to depend on this. but I'm not sure about the best setting, @diemtvu @liminw @philrud any suggestions?

Reply:
1 week, you mean? What happens after the key is expired?

Reply:
Is this the cache expiration time? Making this too large will prevent the key get refresh when necessary, as @quanjielin mentioned.

Reply:
yes, after 1 week, we will try to fetch it from network, and this time if it still failed, we will return a empty public key.

Reply:
@diemtvu In this PR, the refresh job will update the public key independently and will be triggered every 20 minutes to update the key.

Previously, the refresh job only update the public key if it's expired (which is 1 hour), meaning it will make the network request (roughly) every 1 hour even though it's triggered  every 20 minutes.

Reply:
how is ```JwtPubKeyExpireDuration``` used in this PR (after it remove using it from timerjob) ?

Reply:
Just updated the comments, in short, the new code only checks the `expireTime` in `GetPublicKey()` to decide whether or not to fetch it from network. The refresh job will write to it if it gets it from network successfully.